ember-liveagent/src/services/liveagent.ts

```diff
@@ -0,0 +1,157 @@
+import Service from '@ember/service';
+
+declare global {
+  interface Window {
+    embedded_svc?: EmbeddedService;
+  }
+}
+
+interface LiveAgentConfig {
+  salesforceURL: string;
+  communityURL: string;
+  org: string;
+  snapinName: string;
+  baseLiveAgentContentURL: string;
+  deploymentId: string;
+  buttonId: string;
+  baseLiveAgentURL: string;
+  eswLiveAgentDevName: string;
+}
+
+interface BaseSettings {
+  displayHelpButton?: boolean;
+  language?: string;
+  defaultMinimizedText?: string;
+  loadingText?: string;
+  prepopulatedPrechatFields?: Record<string, string>;
+  offlineSupportMinimizedText?: string;
+  enabledFeatures?: string[];
+  entryFeature?: string;
+  storageDomain?: string;
+  extraPrechatFormDetails?: (string | number | boolean)[];
+  extraPrechatInfo?: (string | number | boolean)[];
+  domain: string;
+  isOfflineSupportEnabled: boolean;
+}
+
+interface ConfigOptions {
+  baseLiveAgentContentURL: string;
+  deploymentId: string;
+  buttonId: string;
+  baseLiveAgentURL: string;
+  eswLiveAgentDevName: string;
+  isOfflineSupportEnabled: boolean;
+}
+
+interface EmbeddedService {
+  settings: BaseSettings;
+  init(
+    salesforceURL: string,
+    communityURL: string,
+    gslbBaseURL: string,
+    org: string,
+    string: string,
+    options: ConfigOptions,
+  ): void;
+  hideHelpButton?(): void;
+  showHelpButton?(): void;
+}
+
+interface Options extends LiveAgentConfig {
+  settings: BaseSettings;
+}
+
+export default class LiveAgentService extends Service {
+  isFastBoot = typeof FastBoot !== 'undefined';
+
+  boot(options: Options) {
+    if (this.isFastBoot) {
+      return;
+    }
+    if (!window.embedded_svc) {
+      const { salesforceURL } = options;
+
+      const s = document.createElement('script');
+      s.setAttribute('src', `${salesforceURL}/embeddedservice/5.0/esw.min.js`);
+      s.onload = () => {
+        this._initESW('https://service.force.com', options);
+      };
+      document.body.appendChild(s);
+    } else {
+      this._initESW('https://service.force.com', options);
+    }
+  }
+
+  _initESW(gslbBaseURL: string, options: Options) {
+    if (!window.embedded_svc) {
+      return;
+    }
+
+    const svc = (window.embedded_svc || {}) as EmbeddedService;
```

Comment:
The `embedded_svc` has to be present here (see the guard above), and there is also no fallback case (it would fail on calling `init()`).

```suggestion
    const svc = window.embedded_svc;
```